data/lib/api/user/user_models.dart

```diff
@@ -160,3 +160,178 @@ enum BowlingStyle {
 
   const BowlingStyle(this.value);
 }
+
+enum UserStatType {
+  test(1),
+  other(2);
+
+  final int value;
+
+  const UserStatType(this.value);
+}
+
+@freezed
+class UserStat with _$UserStat {
+  @JsonSerializable(anyMap: true, explicitToJson: true)
+  const factory UserStat({
+    @Default(0) int matches,
+    UserStatType? type,
+    @Default(Batting()) Batting batting,
+    @Default(Bowling()) Bowling bowling,
+    @Default(Fielding()) Fielding fielding,
+  }) = _UserStats;
+
+  factory UserStat.fromJson(Map<String, dynamic> json) =>
+      _$UserStatFromJson(json);
+
+  factory UserStat.fromFireStore(
+    DocumentSnapshot<Map<String, dynamic>> snapshot,
+    SnapshotOptions? options,
+  ) =>
+      UserStat.fromJson(snapshot.data()!);
+}
+
+@freezed
+class Batting with _$Batting {
+  const factory Batting({
+    @Default(0) int innings,
+    @Default(0) int runScored,
+    @Default(0.0) double average,
+    @Default(0.0) double strikeRate,
+    @Default(0) int ballFaced,
+    @Default(0) int fours,
+    @Default(0) int sixes,
+    @Default(0) int fifties,
+    @Default(0) int hundreds,
+    @Default(0) int ducks,
```

Comment:
keep identifier in snake_case here and check at other places specifically in supportCase